sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java

```diff
@@ -221,9 +224,26 @@ private static void unwrapUserCodeException(UserCodeException e) throws Exceptio
    * been finished
    */
   public void processElement(InputT element) throws Exception {
-    if (state == State.FINISHED) {
```

Comment:
It seems like we should still have the `State.FINISHED` check somewhere? Maybe in the `processTimestampedElement` below?